components/api/docker_registry/registry.bal

```diff
@@ -0,0 +1,129 @@
+                error er = error("Failed to extract json payload from getManifest response");
+                return er;
+            }
+        } else if (responseForGetManifest.statusCode == http:OK_200) {
+            log:printDebug(io:sprintf("Successfully retrieved the digest of the atifact \'%s/%s:%s\'", orgName, imageName, artifactVersion));
+            return responseForGetManifest.getHeader("Docker-Content-Digest");
+        } else {
+            error er = error("Failed to fetch the digest of docker manifest. This may be due to an unknown manifest");
+            return er;
+        }
+    } else {
+        error er = error(io:sprintf("Error while fetching digest of docker manifest. %s", responseForGetManifest));
+        return er;
+    }
+}
+
+public function deleteManifest(string orgName, string imageName, string manifestdigest, string bearerToken = "", string userId, string token)
+returns boolean | error {
+    log:printDebug(io:sprintf("Digest received by deleteManifest : \'%s\'", manifestdigest));
+    var responseForDeleteManifest = getResponseFromManifestAPI(orgName, imageName, digest = manifestdigest, bearerToken = bearerToken);
+    if (responseForDeleteManifest is http:Response) {
+        log:printDebug(io:sprintf("Received status code for deleteManifestDigest request: %d", responseForDeleteManifest.statusCode));
+        if (responseForDeleteManifest.statusCode == http:UNAUTHORIZED_401) {
+            var payload = responseForDeleteManifest.getJsonPayload();
+            if (payload is json) {
+                log:printDebug(io:sprintf("Received payload from docker registry for deleteManifest request: %s", payload));
+                string registryScopeForDeleteManifest = buildRegistryScope(payload);
+                log:printDebug(io:sprintf("Registry scope for deleteManifest request: %s", registryScopeForDeleteManifest));
+
+                string tokenToDeleteManifest = getTokenFromDockerAuth(userId, token, registryScopeForDeleteManifest);
+                log:printDebug("Retrived a token to delete manifest");
+
+                return deleteManifest(orgName, imageName, manifestdigest, bearerToken = tokenToDeleteManifest, userId, token);
+            } else {
+                error er = error("Failed to extract json payload from deleteManifest response");
+                return er;
+            }
+        } else if (responseForDeleteManifest.statusCode == http:ACCEPTED_202) {
+            log:printDebug(io:sprintf("Deleted the artifact \'%s/%s\'. Digest : %s", orgName, imageName, manifestdigest));
+            return true;
+        } else {
+            error er = error("Failed to delete the docker manifest. This may be due to an unknown manifest");
+            return er;
+        }
+    } else {
+        error er = error(io:sprintf("Error while deleting docker manifest. %s", responseForDeleteManifest));
+        return er;
+    }
+}
+
+function buildRegistryScope(json payload) returns string {
```

Comment:
Please rename the attribute to something more meaningful such as registryResponsePayload.